packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3891,19 +3893,57 @@ export default abstract class SqlIntegration
     }).join("\n")}`;
   }
 
-  getFilterValues(
-    filterIds: string[],
-    filters: FactFilterInterface[]
-  ): string[] {
-    const filterValues: string[] = [];
-    filterIds.forEach((filterId) => {
-      const filter = filters.find((f) => f.id === filterId);
-      if (filter) {
-        filterValues.push(filter.value);
-      }
-    });
-    return filterValues;
+  public getColumnTopValuesQuery({
+    factTable,
+    column,
+    limit = 50,
+  }: ColumnTopValuesParams) {
+    if (column.datatype !== "string") {
+      throw new Error(`Column ${column.column} is not a string column`);
+    }
+
+    const start = new Date();
+    start.setDate(start.getDate() - 7);
+
+    return format(
+      `
+WITH
+  __factTable as (
+    ${compileSqlTemplate(factTable.sql, {
+      startDate: start,
+      templateVariables: {
+        eventName: factTable.eventName,
+      },
+    })}
+  ),
+  __topValues as (
+    SELECT
+      ${column.column} as value,
+      COUNT(*) as count
+    FROM __factTable
+    WHERE timestamp >= ${this.toTimestamp(start)}
+    GROUP BY ${column.column}
+  )
+${this.selectStarLimit("__topValues ORDER BY count DESC", limit)}
+    `,
+      this.getFormatDialect()
+    );
+  }
+
+  public async runColumnTopValuesQuery(
+    sql: string
+  ): Promise<ColumnTopValuesResponse> {
+    const { rows, statistics } = await this.runQuery(sql);
+
+    return {
+      statistics,
+      rows: rows.map((r) => ({
+        value: r.value + "",
+        count: 1 * r.count,
```

Comment:
Should we prefer this approach going forward over parseFloat and other methods?

Reply:
I'll change this to use `parseFloat`.  It's more clear that way.